src/ts/CanoeHost.ts

```
@@ -0,0 +1,31 @@
+import { inAppelflap } from "ts/PlatformDetection";
+import { AppelflapConnect } from "ts/AppelflapConnect";
+
+export class CanoeHost {
+    /** Tell Appelflap that Canoe is 'locked'
+     * and should not be rebooted */
+    LockCanoe = async (): Promise<string> => {
+        const appelflapConnect = new AppelflapConnect();
+        return await appelflapConnect.lock();
+    };
+
+    StartCanoe = (riotStart: () => void): void => {
+        if (inAppelflap()) {
+            this.LockCanoe()
+                .then((lockResult) => {
+                    if (lockResult !== "ok") {
+                        // The lock didn't succeed, the app is probably still usable.
+                        // We don't yet understand what this means for the user and the app.
+                        // Should we show a small warning?
+                    }
+
+                    riotStart();
+                })
+                .catch(() => {
+                    /* Do nothing */
+                });
```

Why swallow exceptions like this?

---

the whole code is effectively
```
if (inAppelflap()) {
    await this.LockCanoe();
}
riotStart();
```
And you don't even need the riot start, so don't need the await

---

Because we specifically do NOT want any exception here propagating, and therefore we must handle any exceptions even if that means just swallowing them.

This is a Node thing, but it's still general best practice for JS in the browser.

---

I specifically do want to propagate an exception thrown here ( unless you are _expecting_ the exception to happen regularly )